cmd/zpool/zpool_main.c

```diff
@@ -2572,12 +2572,15 @@ show_import(nvlist_t *config)
 				    "incompatibility\n\twhich may cause "
 				    "on-disk corruption with 'zfs recv' and "
 				    "which needs\n\tto be corrected. Enable "
-				    "the bookmark_v2 feature, backup "
-				    "these datasets\n\tto new encrypted "
-				    "datasets, and destroy the old ones. "
-				    "If this pool does\n\tnot contain any "
-				    "encrypted datasets, simply enable the "
-				    "bookmark_v2\n\tfeature.\n"));
+				    "the bookmark_v2 feature. Destroy all "
+				    "snapshots (on encrypted datasets) that "
+				    "were created before enabling bookmark_v2. "
+				    "If preserving those snapshots is "
+				    "required, copy them to new encrypted "
+				    "datasets with zfs send. If this pool "
```

Comment:
this line is worrying me. Existing snapshots won't be able to be `raw`-received even into new encrypted datasets, so effectively one would either have to lose raw-recv capability for these snapshots or copy it. Do we have to make that clearer?

Reply:
I'm not 100% certain, but I thought the approach was to do a non-raw send from $pool/$old_dataset to $pool/$new_dataset. Then the snapshots in $new_dataset were written with an "IVset GUID", and thus everything is fine?

Reply:
Yes, that is fine, that is true. I was just wondering if the requirement/necessity (it will refuse to send raw, but then also sending a -R won't work or a -p for properties because that would force a raw send in the current state of things 😆 if I checked it correctly...

Reply:
so I was wondering whether to change it maybe to
```
If preserving those snapshots is required, zfs send (non-raw) them to new encrypted datasets.
```

Reply:
Sure

Reply:
changed